pkg/clients/azure/dns_test.go

```diff
@@ -0,0 +1,239 @@
+/*
+Copyright 2019 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package azure
+
+import (
+	"fmt"
+	"testing"
+
+	certmanv1alpha1 "github.com/openshift/certman-operator/pkg/apis/certman/v1alpha1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+func TestNewClient(t *testing.T) {
+	t.Run("returns an error if the credentials aren't set", func(t *testing.T) {
+		testClient := setUpTestClient(t, nil)
+
+		_, actual := NewClient(testClient, testHiveAzureSecretName, testHiveNamespace, testHiveResourceGroupName)
+
+		if actual == nil {
+			t.Error("expected an error when attempting to get missing account secret")
+		}
+	})
+
+	t.Run("returns a client if the credentials is set", func(t *testing.T) {
+		testClient := setUpTestClient(t, getAzureSecret(validSecretData))
+
+		_, err := NewClient(testClient, testHiveAzureSecretName, testHiveNamespace, testHiveResourceGroupName)
+
+		if err != nil {
+			t.Errorf("unexpected error when creating the client: %q", err)
+		}
+	})
```

Comment:
these should be set up as table tests, like `TestGetAzureCredentialsFromSecret`, below

Reply:
Done!